src/v0/destinations/marketo_static_list/transform.js

```diff
@@ -71,28 +71,29 @@ const processEvent = (input) => {
     toRemove = getIds(message.properties.listData.remove);
   }
   if (
-    (Array.isArray(toAdd) && toAdd.length > 0) ||
-    (Array.isArray(toRemove) && toRemove.length > 0)
+    (!Array.isArray(toAdd) || toAdd.length === 0) &&
+    (!Array.isArray(toRemove) || toRemove.length === 0)
   ) {
-    if (Array.isArray(toAdd) && toAdd.length > 0) {
-      const payload = batchResponseBuilder(message, Config, token, toAdd, 'add');
+    throw new InstrumentationError(
+      'Invalid leadIds format or no leadIds found neither to add nor to remove',
+    );
+  } else {
+    if (Array.isArray(toRemove) && toRemove.length > 0) {
```

Comment:
Why is the order reversed? toRemove and toAdd order.

Reply:
Making the destination remove calls before the add ones.